src/components/organisms/PostMentions/PostMentions.test.tsx

```diff
@@ -0,0 +1,297 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { PostMentions } from './PostMentions';
+
+// Valid test pubky key (52 lowercase alphanumeric characters)
+const validPubkyKey = 'o1gg96ewuojmopcjbz8895478wdtxtzzber7aezq6ror5a91j7dy';
+
+// Mock @/atoms
+vi.mock('@/atoms', () => ({
+  Link: ({
+    children,
+    href,
+    className,
+    onClick,
+    ...props
+  }: {
+    children: React.ReactNode;
+    href: string;
+    className?: string;
+    onClick?: (e: React.MouseEvent) => void;
+  }) => (
+    <a data-testid="link" href={href} className={className} onClick={onClick} {...props}>
+      {children}
+    </a>
+  ),
+}));
+
+// Mock @/core
+const mockGetDetails = vi.fn();
+vi.mock('@/core', () => ({
+  UserController: {
+    getDetails: (params: { userId: string }) => mockGetDetails(params),
+  },
+}));
+
+// Mock dexie-react-hooks
+vi.mock('dexie-react-hooks', () => ({
+  useLiveQuery: (queryFn: () => Promise<unknown>, deps: unknown[], defaultValue: unknown) => {
+    // Execute the query function to trigger it, but return a controlled value
+    queryFn();
+    // Return the mock value based on what mockGetDetails returns
+    const result = mockGetDetails.mock.results[mockGetDetails.mock.results.length - 1];
+    return result?.value ?? defaultValue;
+  },
+}));
+
+describe('PostMentions', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    mockGetDetails.mockReturnValue(null);
+  });
+
+  describe('Basic rendering', () => {
+    it('renders mention with username when user details are available', () => {
+      mockGetDetails.mockReturnValue({ name: 'TestUser' });
+
+      render(<PostMentions href={`/profile/${validPubkyKey}`}>{`pk:${validPubkyKey}`}</PostMentions>);
+
+      expect(screen.getByText('@TestUser')).toBeInTheDocument();
+    });
+
+    it('renders truncated pubky key when username is not available', () => {
```

Comment:
This test is duplicated later on as `shows truncated key with ellipsis when no username`

Reply:
This wasn't addressed here before merge so I'll do in my new PR.